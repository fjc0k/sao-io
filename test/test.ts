import path from 'path'
import sao from 'sao'

[
  // @index('../src/generators/*', (pp, cc) => `'${pp.name}',`)
  'mp',
  // @endindex
].forEach(generatorName => {
  const generator = path.join(__dirname, '../generators', generatorName)

  test('文件列表正确', async () => {
    const stream = await (sao as any).mock({ generator } as any)
    expect(stream.fileList).toMatchSnapshot()
  })

  test('文件内容正确', async () => {
    const stream = await (sao as any).mock({ generator } as any)
    for (const filePath of stream.fileList) {
      expect(await stream.readFile(filePath)).toMatchSnapshot(filePath)
    }
  })
})
