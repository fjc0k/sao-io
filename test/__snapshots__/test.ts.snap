// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文件列表正确 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  "README.md",
  "global.d.ts",
  "package.json",
  "src/example.ts",
  "tests/test.ts",
  "tsconfig.json",
  "yarn.lock",
]
`;

exports[`文件列表正确 2`] = `
Array [
  "move.json",
  "saofile.ts",
  "template/.editorconfig",
  "template/.gitattributes",
  "template/.gitignore",
  "template/LICENSE",
  "template/README.md",
  "template/global.d.ts",
  "template/package.json",
  "template/src/example.ts",
  "template/tests/test.ts",
  "template/tsconfig.json",
  "template/yarn.lock",
]
`;

exports[`文件列表正确 3`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  "README.md",
  "config/dev.js",
  "config/index.js",
  "config/prod.js",
  "global.d.ts",
  "package.json",
  "project.config.json",
  "sitemap.json",
  "src/app.scss",
  "src/app.tsx",
  "src/assets/index.ts",
  "src/component.ts",
  "src/components/Icon/index.scss",
  "src/components/Icon/index.tsx",
  "src/components/Page/index.scss",
  "src/components/Page/index.tsx",
  "src/components/index.ts",
  "src/pages/Home/Index.scss",
  "src/pages/Home/Index.tsx",
  "src/pages/index.ts",
  "src/services/ListContainer.ts",
  "src/services/index.ts",
  "src/settings.scss",
  "src/store/example.ts",
  "src/store/index.ts",
  "src/utils.ts",
  "tsconfig.json",
  "yarn.lock",
]
`;

exports[`文件列表正确 4`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  "README.md",
  "global.d.ts",
  "package.json",
  "poi.config.js",
  "public/index.html",
  "src/App.scss",
  "src/App.tsx",
  "src/assets/index.ts",
  "src/component.ts",
  "src/components/Icon/index.module.scss",
  "src/components/Icon/index.tsx",
  "src/components/Page/index.module.scss",
  "src/components/Page/index.tsx",
  "src/components/index.ts",
  "src/enhance.ts",
  "src/pages/Home/Index.module.scss",
  "src/pages/Home/Index.tsx",
  "src/pages/Home/Test.tsx",
  "src/pages/index.ts",
  "src/services/ListContainer.ts",
  "src/services/index.ts",
  "src/settings.scss",
  "src/store/example.ts",
  "src/store/index.ts",
  "tests/test.ts",
  "tsconfig.json",
  "yarn.lock",
]
`;
